src/software/sensor_fusion/ssl_protobuf_reader.h

```diff
@@ -37,7 +38,7 @@ class SSLProtobufReader
     Field getFieldData(const SSL_GeometryData &geometry_packet);
```

Comment:
Because this is the only public function of this class for getting `Field`, we actually don't get any of the benefits from the `private` `createFieldFromPacketGeometry` function.

`getFieldData` will return an "invalid" `Field` if the given field geometry doesn't exist or is invalid. SInce we have `Field::isValid` now I think this is fine, since we use that to prevent publishing invalid fields.

So since we don't get any benefit from maintain a `field` member variable, this entire class could be changed to just be regular functions that are responsible from translating protobuf message to datatypes we can use (like `BallDetections`).

Reply:
now that there's no member variables we should be able to remove the class entirely and just leave the functions

Reply:
we should inject dynamic params into it actually